doc/source/signals.rst

```diff
@@ -0,0 +1,192 @@
+Signal API (Experimental)
+=========================
+
+This experimental API allows tasks and actors to generate signals which can
+be received by other tasks and actors. In addition, a task failure or actor
+method failures generate error signals. The error signals enable applications
+to detect failures and potentially recover from failures.
+
+.. autofunction:: ray.experimental.signal.send
+
+Here is a simple example of a task sending a signal.
+
+.. code-block:: python
+
+  import ray.experimental.signal as signal 
+
+  # Define an application level signal.
+  class UserSignal(signal.Signal):
+      def __init__(self, value):
+            self.value = value
+      def get_value(self):
+            return self.value
+
+  # Define a remote function that sends a user signal "value".
+  @ray.remote
+  def send_signal(value):
+      signal.send(UserSignal(value))
+      return
+
+.. autofunction:: ray.experimental.signal.receive
+
+Here is a simple example of receiving signals from and actor or task identified
+by `a`. Note that an actor is identified by its handle, and a task by an object id
+it returns.
+
+.. code-block:: python
+
+  import ray.experimental.signal as signal
```

Comment:
If I call `signal.receive` two times, under what circumstances does the second call only return results from after the first call. E.g., what happens if
1. The two calls happen in the same task.
2. The two calls happen in different tasks on the same actor.
3. The two calls happen in different tasks on the same process.
4. The two calls happen in different processes?
5. The second call happens in a nested task that was submitted by a parent task after the parent task did the first call.